### ISSUE TITLE
Unnecessary PATH injection causes a warning

### ISSUE THREAD

**Issue by janmondry** [ansible/ansible-lint#4212]
##### Summary

There's an issue that causes printing an unnecessary warning and unnecessary injection of a certain path to PATH environmental variable.

https://github.com/ansible/ansible-lint/blob/main/src/ansiblelint/__main__.py#L464
`if (parent / "ansible").exists() and str(parent) not in paths`

Personally I encountered a case when:
- a path in `PATH` ends with a trailing slash (which is correct) (eg. `/app/bin/`), but
- ansible-lint looked for a string without the trailing slash, as returned by `str(parent)` (eg. `/app/bin`),
- which resulted in adding the unnecessary path to `PATH` and printing a warning (example `PATH` could contain something like `/app/bin/:/app/bin`)

Comparing Path objects instead of strings would solve the problem. 

For reference - I created a PR (but couldn't get the checks to work): https://github.com/ansible/ansible-lint/pull/4210

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

not relevant

##### STEPS TO REPRODUCE

Prerequisite: have `ansible` installed in `/app/bin/ansible` and `ansible-lint` in `/app/bin/ansible-lint`

##### Desired Behavior

```paste below
> PATH=/app/bin/ ansible-lint
[...]  # no warning
```

```paste below
> PATH=/app/bin ansible-lint
[...]  # no warning
```

##### Actual Behavior

```paste below
> PATH=/app/bin/ ansible-lint
WARNING: PATH altered to include /app/bin :: This is usually a sign of broken local setup, which can cause unexpected behaviors.
[...]
```

```paste below
> PATH=/app/bin ansible-lint
[...]  # no warning
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
Sorry but nobody does append a slash in PATH elements, we will be closing this as an wontfix because it looks like a user configuration issues.
